src/metpy/plots/wx_symbols.py

```diff
@@ -18,6 +19,67 @@
                                                            'fonts/wx_symbols.ttf'))
 
 
+@exporter.export
+def wx_code_to_numeric(codes):
+    """Determine the numeric weather symbol value from METAR code text.
+
+    A robust method to identifies the numeric value for plotting the correct symbol from a
+    decoded METAR current weather group. The METAR codes should be strings with no missing
+    values or NaN strings (empty strings are okay).
+
+    For example, if from a Pandas Dataframe sfc_df.wxcodes.fillna('')
+
+    Parameters
+    ----------
+    codes : Array like containing string values of METAR weather codes
+
+    Returns
+    -------
+    array of numeric codes of current weather symbols from the wx_code_map for use in
+    plotting.
+    """
+    wx_sym_list = []
+    for s in codes:
+        wxcode = s.split()[0] if ' ' in s else s
+        if len(wxcode) > 4:
```

Comment:
Actually, isn't it possible to just pass even the short codes through here and it will just `KeyError` for the slices that don't make sense?

Reply:
Removed outer if, all the tests passed with the refactor.